compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/input/pointer/SyntheticEventSender.skiko.kt

```diff
@@ -90,7 +90,9 @@ internal class SyntheticEventSender(
             needUpdatePointerPosition = false
 
             previousEvent?.let { event ->
-                if (event.pointers.fastAny { it.down || it.type == PointerType.Mouse }) {
+                // Resent pointer position update only for hover mouse events.
+                // Keep logic in sync with [AndroidComposeView.resendMotionEventOnLayout].
+                if (event.pointers.fastAny { !it.down && it.type == PointerType.Mouse }) {
```

Comment:
```suggestion
                if (event.pointers.fastAny { it.type == PointerType.Mouse }) {
```

for mouse we sent unconditionally before

Reply:
Or is it an attempt to have this logic?

```
 val resend = if (wasMouseEvent) {
                    action != ACTION_HOVER_EXIT && action != ACTION_UP
                } else {
                    action != ACTION_UP
                }
```

Reply:
This logic looks incorrect. The original use case was to hover items on mouse scroll.

Adding `!it.down` check, it doesn't hover in this case:
```
Move
Press
Release
Scroll // Last is !it.down, but we still need send a synthetic
```

Reply:
The goal was to align with the following code:
```
            when (lastEvent.actionMasked) {
                // We currently only care about hover events being updated when layout changes
                ACTION_HOVER_ENTER,
                ACTION_HOVER_MOVE -> {
                    relayoutTime = SystemClock.uptimeMillis()
                    post(resendMotionEventRunnable)
                }
            }
```
Not sure I understand correctly your case. After the `Release` event, the `Scroll` will go with `down == false`, which makes whole condition to be true.

However, I found another bug that can be reproduced like `Move > Press > Scroll` - as a result, it does not send extra `Move` event to remove highlight from an element, if any was highlighted. I tried this code on Android Desktop and it reproduces the same issue.

So, I would keep your suggested solution and remove comments related to "Aligned with Android" because they have bugs in this place.

Reply:
>  the Scroll will go with down == false, which makes whole condition to be true.

Yeah, I had a mistake in the use case. It should be `Move > Press > Scroll`.